composer/core/state.py

```diff
@@ -1118,6 +1118,8 @@ def load_state_dict(
                     if isinstance(serialized_value[metric_name], Metric):
                         # For checkpoints saved using Composer <= 0.13.5
                         serialized_value[metric_name].persistent(mode=True)
+                        # Add new attr in torch2
+                        serialized_value[metric_name]._state_dict_pre_hooks = OrderedDict()
```

Comment:
You can still use torch 2 and hit this on newer Composer. We should check if this exists or not on both sides and add conditionally?

Reply:
I don't think you can actually? Prior to 0.14 (also meaning on torch 1.13.1 or earlier), we save torchmetrics full objects in the state dict. This code path can only be hit if a torchmetrics full object is in the state dict. The issue only occurs if you have loaded a full object, rather than just using the state dict. From 0.14 onwards, we will only save torchmetrics state dict, so it doesn't matter what torch version you are using.

Reply:
lmk if that makes sense